client/blocks/shortcode/frame.js

```diff
@@ -78,14 +80,16 @@ export default class extends React.Component {
 		// `shouldComponentUpdate`
 		const key = Math.random();
 
+		const sandbox = ( this.props.allowSameOrigin ? 'allow-same-origin ' : '' ) + 'allow-scripts';
```

Comment:
seems odd, but the sandbox props follow the same semantic as class names. So you could rewrite this using the `classnames` lib like so:

```js
const sandbox = classnames( {
  "allow-scripts": true,  
  "allow-same-origin": this.props.allowSameOrigin 
} )
```